owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -347,26 +367,79 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
 
     /**
      * OAuth step 1: Get authorization code
+     * Firstly, try the OAuth authorization with Open Id Connect, checking whether there's an available .well-known url
+     * to use or not
      */
-    private fun startOAuthorization() {
+    private fun startOIDCOauthorization() {
         server_status_text.run {
             setCompoundDrawablesWithIntrinsicBounds(R.drawable.progress_small, 0, 0, 0)
             text = resources.getString(R.string.oauth_login_connection)
         }
 
-        mAuthService = OAuthUtils.createAuthorizationService(this)
+        val retrieveConfigurationCallback =
+            RetrieveConfigurationCallback { serviceConfiguration, exception ->
+                if (exception != null) {
+                    Timber.e(exception, "OIDC failed, try with normal OAuth");
```

Comment:
Could you remove that `;` ? This is kotlin 👍

Reply:
![tenor](https://user-images.githubusercontent.com/15261363/79972503-5557ce00-8496-11ea-8509-18070415ca7a.gif)